feat: added `created` hook to directives (#719)
vuejs/docs-next: src/guide/migration/custom-directives.md

In Vue 3, however, we’ve created a more cohesive API for custom directives. As you can see, they differ greatly from our component lifecycle methods even though we’re hooking into similar events. We’ve now unified them like so:

> - **created** - new! This is called before bound element's parent component props are applied

Reviewer: ```suggestion
- **created** - new! This is called before the element's attributes or event listeners are applied.
```
